breathecode/marketing/views.py

```diff
@@ -811,6 +819,90 @@ def delete(self, request, academy_id=None):
         return Response(None, status=status.HTTP_204_NO_CONTENT)
 
 
+class UploadView(APIView):
+    """
+    put:
+        Upload a file to Google Cloud.
+    """
+    parser_classes = [MultiPartParser, FileUploadParser]
+
+    # permission_classes = [AllowAny]
+
+    # upload was separated because in one moment I think that the serializer
+    # not should get many create and update operations together
+    def upload(self, file, academy_id=None, update=False):
+        from ..services.google_cloud import Storage
+
+        result = {
+            'data': [],
+            'instance': [],
+        }
+
+        if not file:
+            raise ValidationException('Missing file in request', code=400)
+
+        # files validation below
+        if file.content_type != MIME_ALLOW:
+            raise ValidationException(f'You can upload only files on the following formats: {MIME_ALLOW}')
+
+        file_bytes = file.read()
+
+        file_name = hashlib.sha256(file_bytes).hexdigest()
+
+        file_bytes = file_bytes.decode('utf-8')
+
+        with open(file.name, 'w') as f:
+            f.write(file_bytes)
+        print('Generic File: ', f)
+        df = pd.read_csv(file.name)
+        os.remove(file.name)
+        print('Data Frame: ', df)
+        required_fields = ['first_name', 'last_name', 'email', 'location', 'phone', 'language']
+
+        # Think about uploading correct files and leaving out incorrect ones
```

Comment:
@alesanchezr the task is creating the `FormEntry` but if the data are duplicated, for example having the same row two times, is possible implement the same behavior of rest of `FormEntry` or validate the duplicated rows and prevent send it